### PR TITLE
Add a new batched AffectedCommits index.

### PR DESCRIPTION
**Pull request by oliverchang**
Part of #951.

### CODE REVIEW

File: osv/impact_test.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        [codecs.encode(c, 'hex') for c in affected_commits[0].commits])
+    self.assertCountEqual(
+        [b'%08d' % i for i in range(10000, 20000)],
+        [codecs.encode(c, 'hex') for c in affected_commits[1].commits])
+    self.assertCountEqual(
+        [b'%08d' % i for i in range(20000, 26000)],
+        [codecs.encode(c, 'hex') for c in affected_commits[2].commits])
+
+  def test_update_deletes_pages(self):
+    """Test deletion of pages."""
```
Comment (michaelkedar) [inline osv/impact_test.py:140]: Unfinished test?

**Review by michaelkedar — COMMENTED** [review]
LGTM, just one thing

File: osv/impact_test.py (hunk 1 above)
Comment (oliverchang) [reply osv/impact_test.py]: Good catch! I didn't need this anymore but I added another test in its place.

File: osv/impact.py [hunk 2]
```diff
@@ -337,23 +338,54 @@ def _throttled_delete(to_delete):
...
+    # commit prior to the fix commit as being affected by a vulnerability.
+    for existing in models.AffectedCommit.query(
+        models.AffectedCommit.bug_id == bug_id):
+      if existing.commit not in commits:
+        to_delete.append(existing.key)
+
+    _throttled_put(to_put)
+    _throttled_delete(to_delete)
+
+  to_put.clear()
```
Comment (another-rex) [inline osv/impact.py:366]: nit: these can be indented to be part of the if statement

**Review by another-rex — APPROVED** [review]
LGTM

**Review by michaelkedar — APPROVED** [review]

File: osv/impact.py (hunk 2 above)
Comment (andrewpollock) [reply osv/impact.py:366]: I'm trying to determine how nitty this actually is. Which if statement?

Comment (oliverchang) [reply osv/impact.py]: Done!

File: osv/impact_test.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+    # These should not be deleted as they're for a different bug.
+    models.AffectedCommits(
+        id='BUG-2-0', bug_id='BUG-2', commits=[], public=True, page=0).put()
+
+    commits = {'%08d' % number for number in range(26000)}
+
+    impact.update_affected_commits('BUG-1', commits, True, write_legacy=False)
+    affected_commits = list(models.AffectedCommits.query())
+    self.assertEqual(4, len(affected_commits))
```
Comment (andrewpollock) [inline osv/impact_test.py:114]: I'm not able to follow why 4 is the expected result here.

I'm seeing 10 pre-populated commits going in for subsequent deletion (BUG-1-{0..9}). 
I'm seeing a single commit for a different bug going in (BUG-2-0).
I'm seeing 26,000 commits for a single bug (BUG-1)

Comment (oliverchang) [reply osv/impact_test.py:114]: These are batched entities. 4 is the expected number that's batched.

**Review by andrewpollock — APPROVED** [review]

**Pull request merged**